src/imfill.jl

```diff
@@ -0,0 +1,44 @@
+"""
+ Image filling Algorithm
+
+ ```
+ filled_img = imfill(img, interval)
+ filled_img = imfill(img, interval, value)
+ filled_img = imfill(img, interval, value, connectivity)
+ ```
+
+ Connected components of an image is found using flood-fill algorithm and returns a copy of
+ the original image after filling objects that falls in the range of interval.
+ For filling objects, represent the holes(part to be filled) with True in your array.
+
+
+ Parameters:
+
+  -  img            = Input image (Boolean array type)
+  -  interval       = objects of size in this range will be filled with False
+  -  connectivity   = connectivity takes the same values as in label_components (Default value is 1:ndims(img))
+
+ """
+
+  function imfill(img::AbstractArray{Bool}, interval::Tuple{Real,Real}, connectivity::Union{Dims, AbstractVector{Int}, BitArray}=1:ndims(img))
+
+     if interval[1] > interval[2] || interval[1] < 0 || interval[2] < 0
+ 	print("Invalid Interval")
```

Comment:
@juliohm I have added some more tests. But they seem to be failing as: `imfill` uses `label_components` from package `Images` and it is not included during testing.
```
ERROR: LoadError: ArgumentError: Package Images not found in current path:
- Run `Pkg.add("Images")` to install the Images package.
```
Can u guide me how to fix this?

Reply:
Good question @aquatiko. In cases like this, we would either migrate `label_components` to somewhere else in the Images.jl project, or migrate `imfill`. I personally like to think of Images.jl as a metapackage that loads the actual submodules of image processing and analysis. In that case, I defend the point that we should migrate `label_components` to somewhere else. What is your opinion on this @timholy? Do you have a suggestion on where we could migrate things so that the project is better structured?

@aquatiko every now and then this happens, and we move things around to achieve a better separation of concepts across packages.

Reply:
@timholy did you have a chance to look into this? Can we migrate `label_components` to somewhere else? Maybe here?

Reply:
Anyone has an opinion on this? What should be the reorganization here? I vote to move `label_components` to some sub-package of `Images.jl`, not sure where it should go though, if here or some new package like `ImageGraphs.jl`

Reply:
Hi Julio,

I am working on bringing the entire Images ecosystem into a new unified API. For an example see: https://github.com/zygmuntszpak/ImageContrastAdjustment.jl. 

I'm also currently working on an `ImageComponentAnalysis` package in collaboration with some great undergraduate students (https://github.com/zygmuntszpak/ImageComponentAnalysis.jl). The current `label_components` would be a natural fit for that package. However, its API would change slightly. It would be associated with the algorithm `Generic()`, that is,  `label_components(Generic(), ...)`. This is to differentiate it from other component labelling algorithms such as contour tracking, i.e. `label_components(ContourTracking(),...)` which are 2D only and assume 8-connectivity.  

The  `ImageComponentAnalysis` is under active development, so not yet ready to be tagged.

Reply:
Lovely @zygmuntszpak ! I am looking forward to see the results. :heart: Do you have the unified API documented somewhere already? I may have missed the recent discussions on the issue tracker.

You mean that the entire Images.jl project is under a major reorganization of concepts?